src/containers/subjects-page/explore-subjects/ExploreSubjects.tsx

```diff
@@ -0,0 +1,101 @@
+import { useState, useCallback } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useSearchParams } from 'react-router-dom'
+
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward'
+import ArrowBackIcon from '@mui/icons-material/ArrowBack'
+
+import useAxios from '~/hooks/use-axios'
+import HashLink from '~/components/hash-link/HashLink'
+import SearchWithFilters from '~/components/search-with-filters/SearchWithFilters'
+import TitleWithDescription from '~/components/title-with-description/TitleWithDescription'
+import AppAutoComplete from '~/components/app-auto-complete/AppAutoComplete'
+
+import { styles } from '~/containers/subjects-page/explore-subjects/ExploreSubjects.styles'
+import { guestRoutes } from '~/router/constants/guestRoutes'
+import { categoryService } from '~/services/category-service'
+import { subjectService } from '~/services/subject-service'
+import { CategoryInterface, SubjectInterface } from '~/types'
+
+const ExploreSubjects = () => {
+  const { t } = useTranslation()
+  const [searchParams, setSearchParams] = useSearchParams()
+  const categoryId = searchParams.get('catId')
+  const categoryIdConvertType = categoryId === 'null' ? null : categoryId
+
+  const [searchValue, setSearchValue] = useState<string>('')
+  const [category, setCategory] = useState<string | null>(categoryIdConvertType)
+
+  const getCategories = useCallback(() => categoryService.getCategories(), [])
+  const getSubjects = useCallback(() => subjectService.getSubjects(), [])
+
+  const { loading: categoriesLoading, response: catogoriesData } = useAxios<CategoryInterface[]>({
+    service: getCategories
+  })
+  const { loading: subjectsLoading, response: subjectsData } = useAxios<SubjectInterface[]>({ service: getSubjects })
+
```

Comment:
Create a custom hook. Add a transform method for mapping the data. useCategoires<Options>({ transform })

const useCategories<T> =...

Reply:
Done